### PR TITLE
Add a CI pipeline to validate elevate with cloudlinux

### PR DESCRIPTION
**Pull request by davelcpanelnet**
Case RE-941: Add a CI pipeline to validate elevate with cloudlinux
 - change trigger event structure
 - update directory layout for multiple OS
 - Increase efficiency of reboot_watch
 - A few hard coded sleeps were added into reboot_watch jobs to get timings right due to cloudlinux taking quite a bit longer than centos runs.

Changelog:

### CODE REVIEW

File: .github/workflows/testsuite.yml [hunk 1]
```diff
@@ -2,12 +2,11 @@ name: testsuite
...
   push:
-    branches:
-      - "*"
-      - '!docs'
     tags-ignore:
-      - "*"
+      - "*"       # Exclude all tags
   pull_request:
+    branches-ignore:
+      - "main"    # Exclude main for PRs as well
```
Comment (toddr) [inline .github/workflows/testsuite.yml:9]: Why are we turning off test runs for the main branch? I want them to run. This seems unrelated to the rest of the commit.

Comment (davelcpanelnet) [reply .github/workflows/testsuite.yml:9]: It seems counterintuitive it just means the testsuite.yml has rules to run instead of the rules here.    Mai just rearranged the rules.

- testsuite will still run against main per the rules in testsuite.yml
- PRs will still run testsuite as a workflow dispatch from within the openstack yml

**Comment by toddr** [comment]
@davelcpanelnet still failing?

**Comment by davelcpanelnet** [comment]
> @davelcpanelnet still failing?

Yes it’s been hitting a failure in stage two where leap bails complaining there is no dnf installed

**Comment by cPholloway** [comment]
Closing in favor of https://github.com/cpanel/elevate/pull/569

#569 slurps in this commit and adds a second commit to fix up a few things in order to make the pipeline pass
